src/ipStore.ts

```diff
@@ -0,0 +1,28 @@
+import fs from 'node:fs'
+
+const storeFile = process.env.STORE_FILE
+
+export const storeIp = (name: string, ip: string): void => {
+  if (!storeFile) {
+    throw new Error('no storeFile!')
+  }
+
+  const state = JSON.parse(fs.readFileSync(storeFile).toString())
+  const existingNames = Object.keys(state)
+  const caseInsensitiveRegexp = new RegExp(name, 'i')
+
+  const matchingName = existingNames.find(
+    name => !!name.match(caseInsensitiveRegexp)
```

Comment:
No need for `!!`

Regex match checks contained instead of exact match

Shadowing names